src/dmd/backend/cgxmm.d

```diff
@@ -726,6 +726,8 @@ opcode_t xmmload(tym_t tym, bool aligned)
         aligned = false;
     switch (tybasic(tym))
     {
+        case TYushort:
```

Comment:
I didn't add the other TYs since it requires extra work. Doing it on a subsequent PR.

Reply:
Well, Buildkite is giving errors on that, somehow. Seem to be failing on register allocation. It seems that this assert doesn't make much sense but I don't have the full picture of the backend to argue. CC @WalterBright can you give an hint on that? It's failing here: https://github.com/dlang/dmd/blob/727606c0b5bc7610ddf7d6f49dd50ca0433bdd38/src/dmd/backend/cod1.d#L1932 .

The way to reproduce this is:
```
float bug23077(float x) {
    byte i = *cast(byte*)&x;
    ++i;
    return *cast(float*)&i;
}
```
And add `TYbyte` type to the xmmload/xmmstore functions. There is obviously another way to reproduce this as the code doesn't have `TYbyte` right now, but this reproduction seem to trigger the same code.